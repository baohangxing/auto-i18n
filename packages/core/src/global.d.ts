declare module '@babel/plugin-proposal-optional-chaining';
declare module '@babel/plugin-syntax-jsx';
declare module '@babel/plugin-proposal-optional-chaining';
declare module '@babel/plugin-syntax-class-properties';
declare module '@babel/plugin-syntax-decorators';
declare module '@babel/plugin-syntax-object-rest-spread';
declare module '@babel/plugin-syntax-async-generators';
declare module '@babel/plugin-syntax-do-expressions';
declare module '@babel/plugin-syntax-dynamic-import';
declare module '@babel/plugin-syntax-export-extensions';
declare module '@babel/plugin-syntax-function-bind';
declare module '@babel/preset-typescript'; 
