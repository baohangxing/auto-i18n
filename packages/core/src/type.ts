export interface I18nCallRule {
  /** caller of functionName
       *  @default ''
      */
  caller: string
  /** function name of `t`
       *  @default 't'
      */
  functionName: string

  definedDeclaration: string
  /**
       * import i18n t Declaration
       */
  importDeclaration: string
}

export type FileExtension = 'js' | 'ts' | 'cjs' | 'mjs' | 'jsx' | 'tsx' | 'vue'

export interface AutoConfig {
  /**
      * local Locales-Json directory Paths, all user json file should found in this paths
      *
      * @example [`/Users/XXX/GM/src/lang/locales`]
      */
  localesJsonDirs: string[] | string

  /**
       * locales used, the locale will not trans if not in `locales`
       *
       * @example [`ja-jp`, `zh-cn`]
       */
  locales: string[]

  /**
       * the base locale in your project, make sure it is in `locales` and all words has configured in locale-Json file
       *
       * @default `zh-cn`
       */
  baseLocale: string

  /**
       *
       * @param locale
       * @returns
       */
  untransSymbol: (locale: string) => string

  /**
 * translate word rule when not translated
 *
 * @param word
 * @param locale the word current locale, your baseLocale
 * @param toLocale to translate locale like `ja-jp`
 * @returns
 */
  transLacaleWord?: (word: string, locale: string, toLocale: string) => Promise<string>

  /**
      * files to auto trans, using fast-glob rule
      *
      * see deteils: https://github.com/mrmlnc/fast-glob
      *
      * @example ["src/*.{vue}", "view/**"]
      */
  includes: string[]

  /**
       * exclude files to auto trans, using fast-glob rule
       *
       * see deteils: https://github.com/mrmlnc/fast-glob
       *
       * @example  ['**\/node_modules/**\/*']
       */
  exclude: string[]

  /**
       * output filed directory path,
       * the default output filed directory path is your project path
       *
       * @default `./ `
       */
  outputFileDir: string

  i18nCallRules: Record<FileExtension, I18nCallRule>
}

export interface TransCommandOption {
  filePath: string
  generateNewFile: boolean
  newFileName: string
}

export interface LangJson {
  name: string
  path: string
}
