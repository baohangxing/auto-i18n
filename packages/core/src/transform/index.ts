import type { FileExtension, I18nCallRules } from '../types'
import log from '../utils/log'
import transformJs from './transformJs'
import transformVue from './transformVue'
import { initJsParse, initTsxParse } from './parse'

const transform = (
  code: string,
  ext: FileExtension,
  rules: I18nCallRules,
  replace = true,
): {
  code: string
} => {
  switch (ext) {
    case 'cjs':
    case 'mjs':
    case 'js':
    case 'jsx':
      return transformJs(code, {
        rule: rules[ext],
        parse: initJsParse(),
      }, replace)
    case 'ts':
    case 'tsx':
      return transformJs(code, {
        rule: rules[ext],
        parse: initTsxParse(),
      }, replace)
    case 'vue':
      return transformVue(code, rules[ext], replace)
    default:
      log.error(`not support transform .${ext} extension`)
      return { code }
  }
}

export { transform }
