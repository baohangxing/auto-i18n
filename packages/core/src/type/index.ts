export * from './command'
export * from './autoConfig'
