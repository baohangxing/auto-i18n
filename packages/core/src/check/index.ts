import { createRequire } from 'module'
import type { NodePath } from '@babel/traverse'
import type {
  CallExpression,
} from '@babel/types'
import type * as traverseType from '@babel/traverse/index'
import { parse } from '@vue/compiler-sfc'
import * as htmlparser2 from 'htmlparser2'
import mustache from 'mustache'
import type { ParseResult } from '@babel/core'
import type { I18nCallRule, Log } from '../types'
import { IGNORE_REMARK } from '../config/constants'
import { trimValue } from '../transform/tools'
import { initTsxParse } from '../transform/parse'

interface GetI18nKeysOptions {
  rule: I18nCallRule
  parse: (code: string) => ParseResult | null

  logger?: Log<any>
}

const require = createRequire(import.meta.url)
// see https://github.com/babel/babel/issues/13855
const traverse: typeof traverseType.default = require('@babel/traverse').default

const getJsI18nKeys = (code: string, options: GetI18nKeysOptions): string[] => {
  const rule = options.rule

  const res = new Set<string>()

  const checkAST = (code: string, options: GetI18nKeysOptions) => {
    function getTraverseOptions() {
      return {

        CallExpression(path: NodePath<CallExpression>) {
          const { node } = path
          const { transCaller, transIdentifier } = rule
          const callee = node.callee

          let mayIsArguments = false

          // 无调用对象的情况，例如 t('xx')
          if (!transCaller && callee.type === 'Identifier' && callee.name === transIdentifier)
            mayIsArguments = true

          // 有调用对象的情况，例如this.$t('xx')、i18n.$t('xx')
          if (callee.type === 'MemberExpression') {
            if (callee.property && callee.property.type === 'Identifier') {
              if (callee.property.name === transIdentifier) {
                // 处理形如i18n.$t('xx')的情况
                if (callee.object.type === 'Identifier' && callee.object.name === transCaller)
                  mayIsArguments = true

                // 处理形如this.$t('xx')的情况
                if (callee.object.type === 'ThisExpression' && transCaller === 'this')
                  mayIsArguments = true
              }
            }
          }

          if (mayIsArguments && node.arguments[0]) {
            if (node.arguments[0].type === 'StringLiteral')
              res.add(node.arguments[0].value)

            if (node.arguments[0].type === 'TemplateLiteral'
              && node.arguments[0].expressions.length === 0
              && node.arguments[0].quasis?.[0].value.raw !== '')
              res.add(node.arguments[0].quasis[0].value.raw)
          }
        },
      }
    }

    const ast = options.parse(code)
    traverse(ast, getTraverseOptions())
    return ast
  }

  checkAST(code, options)

  return [...res]
}

const getJsSyntaxI18nKeys = (source: string, options: GetI18nKeysOptions): string[] => {
  if (source.startsWith('{') && source.endsWith('}'))
    source = `___temp = ${source}`

  return getJsI18nKeys(source,
    {
      rule: {
        ...options.rule,
        variableDeclaration: '',
        importDeclaration: '',
      },
      parse: initTsxParse(),
      logger: options.logger,
    },
  )
}

const getTemplateKeys = (code: string, options: GetI18nKeysOptions): string[] => {
  const res: string [] = []

  let shouldIgnore = false
  const parser = new htmlparser2.Parser(
    {
      onopentag(tagName, attributes) {
        if (shouldIgnore)
          return

        for (const key in attributes) {
          const attrValue = attributes[key]
          // v-for 的 vue自定义语法无法用 getJsSyntaxI18nKeys
          const isVueDirective = key.startsWith(':') || key.startsWith('@')
            || (key.startsWith('v-') && !key.startsWith('v-for'))
          if (isVueDirective) {
            const keys = getJsSyntaxI18nKeys(attrValue, options)
            res.push(...keys)
          }
        }
      },

      ontext(text) {
        if (shouldIgnore)
          return

        const tokens = mustache.parse(text)
        for (const token of tokens) {
          const type = token[0]
          let value = token[1]

          value = trimValue(value)
          if (type === 'name') {
            const keys = getJsSyntaxI18nKeys(value, options)
            res.push(...keys)
          }
        }
      },
      onclosetag() {
        shouldIgnore = false
      },
      oncomment(comment) {
        if (comment.includes(IGNORE_REMARK))
          shouldIgnore = true
      },
    },
    {
      lowerCaseTags: false,
      recognizeSelfClosing: true,
      lowerCaseAttributeNames: false,
    },
  )

  parser.write(code)
  parser.end()
  return res
}

const getScriptKeys = (source: string, options: GetI18nKeysOptions): string[] => {
  return getJsI18nKeys(source, options)
}

const getVueI18nKeys = (
  code: string,
  options: GetI18nKeysOptions,
): string[] => {
  const { descriptor, errors } = parse(code)
  if (errors.length > 0) {
    options.logger?.error('Parse vue error', errors)
    return []
  }

  const res = new Set<string>()

  const { template, script, scriptSetup } = descriptor

  if (template) {
    const keys = getTemplateKeys(template.content, options)
    for (const x of keys)
      res.add(x)
  }

  if (script) {
    const keys = getScriptKeys(script.content, options)
    for (const x of keys)
      res.add(x)
  }

  if (scriptSetup) {
    const keys = getScriptKeys(scriptSetup.content, options)
    for (const x of keys)
      res.add(x)
  }

  return [...res]
}

export { getJsI18nKeys, getVueI18nKeys }
export type { GetI18nKeysOptions }
