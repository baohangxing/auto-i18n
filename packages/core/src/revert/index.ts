import { createRequire } from 'module'
import type { NodePath } from '@babel/traverse'
import type {
  CallExpression,
} from '@babel/types'
import type * as traverseType from '@babel/traverse/index'
import type { ParseResult } from '@babel/core'
import type { SFCScriptBlock, SFCTemplateBlock } from '@vue/compiler-sfc'
import { parse } from '@vue/compiler-sfc'
import * as htmlparser2 from 'htmlparser2'
import mustache from 'mustache'
import presetTypescript from '@babel/preset-typescript'
import fsExtra from 'fs-extra'
import type * as generatorType from '@babel/generator/index'
import ejs from 'ejs'
import { getJsonPath } from '../config/config'
import type { I18nCallRule, Log } from '../types'
import { IGNORE_REMARK, VUE_COMMENT_TYPE } from '../config/constants'
import { getFileComment, getStringLiteral, getWrapperTemplate, mergeCode } from '../transform/tools'
import { initJsParse, initTsxParse } from '../transform/parse'
import { getValueByKey } from '../utils/help'

type Handler = (source: string, options: RevertOptions,) => string

interface RevertOptions {
  rule: I18nCallRule
  parse: (code: string) => ParseResult | null
  locale: string
  /**
   * Whether handle js in vue
   */
  isJsInVue?: boolean

  loger?: Log<any>
}

const require = createRequire(import.meta.url)
// see https://github.com/babel/babel/issues/13855
const traverse: typeof traverseType.default = require('@babel/traverse').default
const babelGenerator: typeof generatorType.default = require('@babel/generator').default

const revertWordByKey = (locale: string) => {
  const { baseLangJson, otherLangJsons } = getJsonPath()

  let langObj: any = fsExtra.readJsonSync(baseLangJson.path)

  for (const x of otherLangJsons) {
    if (x.name === locale)
      langObj = fsExtra.readJsonSync(x.path)
  }
  return (key: string) => {
    const val = getValueByKey(langObj, key)
    return typeof val === 'string' ? val : ''
  }
}

const revertJs = (code: string, options: RevertOptions): string => {
  const rule = options.rule

  const getWordByKey = revertWordByKey(options.locale)

  const revertAST = (code: string, options: RevertOptions) => {
    function getTraverseOptions() {
      return {

        CallExpression(path: NodePath<CallExpression>) {
          const { node } = path
          const { transCaller, transIdentifier } = rule
          const callee = node.callee

          if (!transCaller && callee.type === 'Identifier' && callee.name === transIdentifier) {
            if (node.arguments[0] && node.arguments[0].type === 'StringLiteral')
              node.arguments[0] = getStringLiteral(getWordByKey(node.arguments[0].value))
            // TODO 增加参数的替换
            // TODO 去掉 callee
          }

          if (callee.type === 'MemberExpression') {
            // TODO 去掉 transCaller
            if (callee.property && callee.property.type === 'Identifier') {
              if (callee.property.name === transIdentifier) {
                if (callee.object.type === 'Identifier' && callee.object.name === transCaller) {
                  if (node.arguments[0] && node.arguments[0].type === 'StringLiteral')
                    node.arguments[0] = getStringLiteral(getWordByKey(node.arguments[0].value))
                }
                if (callee.object.type === 'ThisExpression' && transCaller === 'this') {
                  if (node.arguments[0] && node.arguments[0].type === 'StringLiteral')
                    node.arguments[0] = getStringLiteral(getWordByKey(node.arguments[0].value))
                }
              }
            }
          }
        },

      }
    }

    const ast = options.parse(code)
    traverse(ast, getTraverseOptions())
    return ast
  }

  const ast = revertAST(code, options)

  const result = (!ast ? code : babelGenerator(ast)) as generatorType.GeneratorResult

  return result.code
}

const revertJsSyntax = (source: string, options: RevertOptions): string => {
  let isObjectStruct = false
  if (source.startsWith('{') && source.endsWith('}')) {
    isObjectStruct = true
    source = `___temp = ${source}`
  }
  const code = revertJs(source, {
    rule: {
      ...options.rule,
      variableDeclaration: '',
      importDeclaration: '',
    },
    parse: initTsxParse(),
    loger: options.loger,
    locale: options.locale,
  })

  let stylizedCode = code

  if (isObjectStruct)
    stylizedCode = stylizedCode.replace('___temp = ', '')

  stylizedCode = stylizedCode.replace(/;$/gm, '')
  return stylizedCode.endsWith('\n') ? stylizedCode.slice(0, stylizedCode.length - 1) : stylizedCode
}

const revertTemplate = (code: string, options: RevertOptions): string => {
  let htmlString = ''

  let shouldIgnore = false

  const parser = new htmlparser2.Parser(
    {
      onopentag(tagName, attributes) {
        let attrs = ''
        if (shouldIgnore) {
          for (const key in attributes) {
            const attrValue = attributes[key]
            if (attrValue !== '') {
              attrs += ` ${key}="${attrValue}"`
            }
            else {
              if (key.match(/^[@]/)) {
                options.loger?.verbose(`pleace check the changing of attributes: ${key}="${attrValue}"`)
                attrs += ` ${key}="${attrValue}"`
              }
              else {
                options.loger?.verbose(`pleace check the changing of attributes: ${key}`)
                attrs += ` ${key}`
              }
            }
          }
          htmlString += `<${tagName}${attrs}>`
          return
        }
        for (const key in attributes) {
          const attrValue = attributes[key]
          const isVueDirective = key.startsWith(':') || key.startsWith('@')
            || (key.startsWith('v-') && !key.startsWith('v-for'))
          if (isVueDirective) {
            const source = revertJsSyntax(attrValue, options)
            attrs += ` ${key}="${source}"`
          }
          else if (attrValue === '') {
            if (key.match(/^[@]/)) {
              options.loger?.verbose(`pleace check the changing of attributes: ${key}="${attrValue}"`)
              attrs += ` ${key}="${attrValue}"`
            }
            else {
              options.loger?.verbose(`pleace check the changing of attributes: ${key}`)
              attrs += ` ${key}`
            }
          }
          else {
            attrs += ` ${key}="${attrValue}"`
          }
        }
        htmlString += `<${tagName}${attrs}>`
      },

      ontext(text) {
        if (shouldIgnore) {
          htmlString += text
          return
        }

        let str = ''
        const tokens = mustache.parse(text)
        for (const token of tokens) {
          const type = token[0]
          const value = token[1]

          if (type === 'text') {
            str += value
          }
          else if (type === 'name') {
            const source = revertJsSyntax(value, options)
            str += `{{${source}}}`
          }
          else if (type === VUE_COMMENT_TYPE) {
            // 形如{{!xxxx}}这种形式，在mustache里属于注释语法
            str += `{{!${value}}}`
          }
        }

        htmlString += str
      },
      onclosetag(tagName, isImplied) {
        shouldIgnore = false
        if (isImplied) {
          htmlString = `${htmlString.slice(0, htmlString.length - 1)}/>`
          return
        }
        htmlString += `</${tagName}>`
      },

      oncomment(comment) {
        if (comment.includes(IGNORE_REMARK))
          shouldIgnore = true

        htmlString += `<!--${comment}-->`
      },
    },
    {
      lowerCaseTags: false,
      recognizeSelfClosing: true,
      lowerCaseAttributeNames: false,
    },
  )

  parser.write(code)
  parser.end()
  return htmlString
}

const revertScript = (source: string, options: RevertOptions): string => {
  return revertJs(source, {
    rule: options.rule,
    isJsInVue: true,
    loger: options.loger,
    parse: initJsParse([[presetTypescript,
      { isTSX: true, allExtensions: true }]]),
    locale: options.locale,
  })
}

const generateSource = (
  sfcBlock: SFCTemplateBlock | SFCScriptBlock,
  handler: Handler,
  options: RevertOptions,
): string => {
  const wrapperTemplate = getWrapperTemplate(sfcBlock)
  const source = handler(sfcBlock.content, options)
  return ejs.render(wrapperTemplate, {
    code: source,
  })
}

const revertVue = (
  code: string,
  options: RevertOptions,
): string => {
  const { descriptor, errors } = parse(code)
  if (errors.length > 0) {
    options.loger?.error('Parse vue error', errors)
    return ''
  }

  const { template, script, scriptSetup, styles } = descriptor
  let templateCode = ''
  let scriptCode = ''
  let stylesCode = ''
  let scriptSetupCode = ''
  const fileComment = getFileComment(descriptor)

  if (template)
    templateCode = generateSource(template, revertTemplate, options)

  if (script)
    scriptCode = generateSource(script, revertScript, options)

  if (scriptSetup)
    scriptSetupCode = generateSource(scriptSetup, revertScript, options)

  if (styles) {
    for (const style of styles) {
      const wrapperTemplate = getWrapperTemplate(style)
      const source = style.content
      stylesCode
        += `${ejs.render(wrapperTemplate, {
          code: source,
        })}\n`
    }
  }

  code = mergeCode(templateCode, scriptCode, scriptSetupCode, stylesCode)
  if (fileComment)
    code = `${fileComment}\n${code}`

  return code
}

export { revertJs, revertVue }
export type { RevertOptions }
