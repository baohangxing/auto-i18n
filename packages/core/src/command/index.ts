import { Command, Option } from 'commander'
import { CLI_CONFIG_NAME } from '../config/constants'
import type { RevertCommandOption, TransCommandOption } from '../types/config'
import { checkAllTranslated } from '../check'
import { revert } from '../revert'
import { version } from './../../package.json'
import { generateXlsx } from './genXlsx'
import { init } from './init'
import { trans } from './trans'
import { updateLocales, updateLocalesFromXlsx } from './update'

const program = new Command()

program
  .name(CLI_CONFIG_NAME)
  .description('A CLI to help you transform your project '
    + 'to an internationalization project automatically.')
  .version(version)

program.addOption(new Option('-d, --debug').hideHelp())
program.addOption(new Option('-vb, --verbose').hideHelp())

program.on('option:verbose', () => {
  process.env.AUTO_I18N_VERBOSE = program.opts().verbose
})
program.on('option:debug', () => {
  process.env.AUTO_I18N_DEBUG = program.opts().debug
})

program.command('init')
  .description('Init yo-auto-i18n, generate an new default config file automatically.')
  .action(() => {
    init()
  })

program.command('trans')
  .description('Transform a single file or all files.')
  .argument('[transPath]', 'file or directory path')
  .option('--modify', 'whether to modify on the source file')
  .option('-t, --template <string>', 'template json file which inclues all language key')
  .action((transPath, options) => {
    const transCommandOption: TransCommandOption = {
      transPath: transPath ?? '',
      modifyMode: !!options.modify,
      templateFile: options.template ?? '',
    }
    trans(transCommandOption)
  })

program.command('update')
  .description('Update other locales Json files by base locales Json file')
  .option('-t, --templateXlsx <string>', 'update locales json files by the Xlsx template file,'
    + ' the frirt line of the first sheet should have all locales names and the `key`', '')
  .action(async (options) => {
    await updateLocales()
    options.templateXlsx && (await updateLocalesFromXlsx(options.templateXlsx))
  })

program.command('genlsx')
  .description('Generate an Xlsx file by locales json files,'
    + ' the frirt line of in the first sheet of the xlsx will have all locales-name and the `key`')
  .action(() => {
    generateXlsx()
  })

program.command('check')
  .description('Check weather all word has been transformed or not')
  .action(() => {
    checkAllTranslated()
  })

program.command('revert')
  .description('revert a transformed file or files in a directory')
  .argument('<revertPath>', 'file or directory path')
  .option('-t, --target <string>', 'the target language when revert')
  .action(async (revertPath, options) => {
    const revertCommandOption: RevertCommandOption = {
      revertPath: revertPath ?? '',
      target: options.target ?? '',
    }
    await revert(revertCommandOption)
  })

program.parse()
