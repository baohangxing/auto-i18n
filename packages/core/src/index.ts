import './command'
export * from './types'
