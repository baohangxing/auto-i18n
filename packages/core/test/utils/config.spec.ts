import path from 'path'
import fs from 'fs'
import { afterAll, describe, expect, it } from 'vitest'
import serialize from 'serialize-javascript'
import type { LangJson } from '../../src'
import { CONFIG_FILE_NAME, defaultAutoBaseConfig, getAutoBaseConfig, getJsonPath } from '../../src'
import { playground } from '../_helps/playground'
import { trimRootPath } from '../_helps/utils'

const filePath = path.join(process.cwd(), CONFIG_FILE_NAME)

describe('#config', () => {
  describe('getAutoBaseConfig', () => {
    it('should get default config', async () => {
      let res
      await playground(async () => {
        const config = defaultAutoBaseConfig
        const content = `module.exports = ${serialize(config, {
          unsafe: true,
        })}`

        fs.writeFileSync(filePath, content, 'utf8')
        res = getAutoBaseConfig()
      })
      expect(res).toMatchSnapshot()
    })

    it('should get default config by file path', async () => {
      await playground(async () => {
        expect(getAutoBaseConfig(defaultAutoBaseConfig, filePath)).toMatchSnapshot()
      })
    })
  })

  describe('getJsonPath', () => {
    it('should get none JSON paths', async () => {
      let baseLangJsonRes, otherLangJsonsRes

      await playground(async () => {
        const {
          baseLangJson,
          otherLangJsons,
        } = getJsonPath()
        baseLangJsonRes = baseLangJson
        otherLangJsonsRes = otherLangJsons
        console.log(111, otherLangJsons)
      })
      expect(baseLangJsonRes).toBe(undefined)
      expect(otherLangJsonsRes).toEqual([])
    })

    it('should get JSON paths', async () => {
      let baseLangJsonRes: LangJson | undefined
      let otherLangJsonsRes: LangJson[] = []

      let nameRes = ''
      await playground(async ({ name }) => {
        const {
          baseLangJson,
          otherLangJsons,
        } = getJsonPath()
        baseLangJsonRes = baseLangJson
        otherLangJsonsRes = otherLangJsons
        nameRes = name
      })
      expect(baseLangJsonRes).not.toBe(undefined)
      expect(trimRootPath([baseLangJsonRes?.path?.replace(nameRes, '__name') ?? ''])).toMatchSnapshot()
      expect(trimRootPath(otherLangJsonsRes.map(x => x.path.replace(nameRes, '__name')))).toMatchSnapshot()
    })
  })
})

afterAll(async () => {
  if (fs.existsSync(filePath))
    fs.rmSync(filePath)
})
