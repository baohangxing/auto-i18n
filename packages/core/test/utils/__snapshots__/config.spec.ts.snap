// Vitest Snapshot v1

exports[`#config > getAutoBaseConfig > should get default config 1`] = `
{
  "baseLocale": "",
  "i18nCallRules": {
    "cjs": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
    "js": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
    "jsx": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
    "mjs": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
    "ts": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
    "tsx": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
    "vue": {
      "importDeclaration": "import { useI18n } from \\"vue-i18n\\";",
      "transCaller": "",
      "transIdentifier": "t",
      "variableDeclaration": "const { t } = useI18n();",
    },
  },
  "locales": [],
  "localesJsonDirs": [
    "**/locales/**.json",
    "!test/**",
  ],
  "transInterpolationsMode": "NamedInterpolationMode",
}
`;
